drivers/storage/portworx/util/util.go

```diff
@@ -1449,17 +1491,20 @@ func CountStorageNodes(
 		}
 
 		// In case of non metro-DR setup, all portworx nodes that contain backend storage in the enumerate response are storage nodes
-		// In the case of metro DR setup, temporarily using existing logic to determine storage nodes count
-		// TODO: Need to update this logic in metro DR setup to use portworx nodes in the current domain
+		// In the case of metro DR setup, for PX versions 3.1.1 onwards, px nodes of that cluster are counted as storage nodes
+		// When the DR cluster uses px version lesser than 3.1.1, only the nodes that are part of the current k8s cluster are counted as storage nodes
 		if isQuorumMember {
 			if !isDRSetup {
 				storageNodesCount++
-			} else {
+			} else if useClusterDomain {
+
 				if node.ClusterDomain == currentClusterDomain {
 					storageNodesCount++
 				} else {
-					logrus.Debugf("node %s is not part of cluster domain %s ", node.SchedulerNodeName, currentClusterDomain)
+					logrus.Debugf("node %s is not part of cluster domain %s ", node.Id, currentClusterDomain)
```

Comment:
nit: Can we print both id and name?
```
logrus.Debugf("node %s (%s) is not part of the cluster domain %s ", node.SchedulerNodeName, node.Id, currentClusterDomain)
```

Reply:
Sometimes the node will not have a schedulername right, will that be fine ?

Reply:
Yes, it is fine if it empty. We are making a change in porx to always have the scheduler name anyway.